synapse/federation/sender/__init__.py

```diff
@@ -355,6 +355,39 @@ async def handle_event(event: EventBase) -> None:
                     if not is_mine and send_on_behalf_of is None:
                         return
 
+                    # We also want to not send out-of-band membership events.
```

Comment:
Is `out-of-band` membership a spec-defined or Synapse-internal term? (Seems to be the latter.) Is there a reference for what they are somewhere?

The 3.5 cases below seem to enumerate them and explain why we don't want to send them out. It might be useful to have this list of 3.5 event types here (a more authoritative place?):

https://github.com/matrix-org/synapse/blob/c027bc0e4b071d3d40c7b0c1e7011ad5c8c3d0a0/synapse/events/__init__.py#L214-L223

Reply:
> Is `out-of-band` membership a spec-defined or Synapse-internal term? (Seems to be the latter.) Is there a reference for what they are somewhere?

https://matrix-org.github.io/synapse/develop/development/room-dag-concepts.html#out-of-band-membership-events has some info on it.

Reply:
Thanks, that makes sense. Perhaps just the introductory sentence from https://matrix-org.github.io/synapse/develop/development/room-dag-concepts.html#out-of-band-membership-events

> A special case of outlier events are some membership events for federated rooms that we aren't full members of. 

would help illuminate `is_out_of_band_membership`? Not going to block this on that though.

Reply:
I've updated the docstring for `is_out_of_band_membership` in 947f651ab3.